eclair-core/src/main/scala/fr/acinq/eclair/channel/InteractiveTxBuilder.scala

```diff
@@ -440,17 +442,25 @@ private class InteractiveTxBuilder(replyTo: ActorRef[InteractiveTxBuilder.Respon
   private def getInputDetails(txIn: TxIn, currentInputs: Seq[TxAddInput]): Future[Either[UnusableInput, TxAddInput]] = {
     currentInputs.find(i => txIn.outPoint == toOutPoint(i)) match {
       case Some(previousInput) => Future.successful(Right(previousInput))
-      case None => wallet.getTransaction(txIn.outPoint.txid).map(previousTx => {
-        if (Transaction.write(previousTx).length > 65000) {
-          // Wallet input transaction is too big to fit inside tx_add_input.
-          Left(UnusableInput(txIn.outPoint))
-        } else if (!Script.isNativeWitnessScript(previousTx.txOut(txIn.outPoint.index.toInt).publicKeyScript)) {
-          // Wallet input must be a native segwit input.
-          Left(UnusableInput(txIn.outPoint))
-        } else {
-          Right(TxAddInput(fundingParams.channelId, UInt64(0), previousTx, txIn.outPoint.index, txIn.sequence))
+      case None =>
+        val inputTxDetails = for {
+          tx <- wallet.getTransaction(txIn.outPoint.txid)
+          confirmations <- if (fundingParams.requireConfirmedInputs.local) wallet.getTxConfirmations(txIn.outPoint.txid) else Future.successful(None)
```

Comment:
`getrawtransaction` with `debug=true` returns transaction details + confirmations, but it's not currently supported in `BitcoinCoreClient` and probably not very important to optimize.